### DISCUSSION TITLE
CloudFront API supported versions

### DISCUSSION THREAD

**Discussion by wsamoht** [aws/aws-sdk-php#2783] [General]
I am wondering if anyone knows where supported API versions are listed for AWS services and where announcements are made of deprecations/retirements. At the moment I am wondering about CloudFront.

I ask because I just updated the PHP SDK to the latest version 3.281.15 and I started getting errors when creating a new `CloudFrontClient`. I went back one version at a time and discovered the error started happening in 3.281.13.

I originally followed the examples [here](https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/cloudfront-example-signed-url.html) for signing URLs which uses version `2014-11-06`.

```php
new CloudFrontClient([
    "profile" => "default",
    "version" => "2014-11-06",
    "region" => "us-east-1"
]);
```
This now throws the following error:
```
TYPE ERROR  Aws\EndpointV2\EndpointProviderV2::__construct(): Argument #1 ($ruleset) must be of type array, null given, called in vendor/aws/aws-sdk-php/src/ClientResolver.php on line 1049
```

Updating to the latest version `2020-05-31` it started working again.

**EDIT**
I guess this shows supported versions https://docs.aws.amazon.com/aws-sdk-php/v3/api/class-Aws.CloudFront.CloudFrontClient.html for at least the SDK. But, is there a list somewhere for the API itself? And is there somewhere for announcements of retired versions?

**Comment by wsamoht** [comment]
I created a PR to update the docs [here](https://github.com/awsdocs/aws-doc-sdk-examples/pull/5435).

**Comment by stobrien89** [comment]
Hi @wsamoht,

I spoke too soon in my PR— what happened is that our [manifest file](https://github.com/aws/aws-sdk-php/blob/79a3ed5bb573f592823f8b1cffe0dbac3132e6b4/src/data/manifest.json#L425) maps this api version to another one that was unintentionally modified (emptied) in a recent model sync.  I think your example changes are still worth accepting, because at this point we recommend using the latest version of a service's api model.  This should be fixed in tomorrow's release at the latest if you'd like to switch back to the older version.

**Reply by wsamoht** [reply]
I was thinking it might have been an unintentional change but nothing stood out comparing between the SDK versions :)

The latest API version works for me as we are only using the `CloudFrontClient` for signing URLs which works just fine.

Thanks for taking a look!
